rust/toolchain.bzl

```diff
@@ -221,6 +241,8 @@ rust_toolchain = rule(
         "_crosstool": attr.label(
             default = Label("//tools/defaults:crosstool"),
         ),
+        "opt_level": attr.string_dict(default = {"opt": "3", "dbg": "0", "fastbuild": "0"}),
```

Comment:
Optional Nit:

I find the interchanging use of rust's opt and dbg and bazel's "opt" and "db" to be very confusing here.

Are you open to making this more verbose: "compilation_mode_to_opt_level" and "compilation_mode_to_debug_info_level"?

Reply:
How about naming the attributes `rust_opt_level` and `rust_debug_info`, respectively? I'm not opposed to the verbose names you suggest, but I'm also not bothered by the current scheme either :).

Reply:
I come from a Java context, so I don't feel sufficiently spoon fed when `rust_opt_level` contains something that isn't a `string` or `num`. The map implies some kind of relationship or nuance that is hidden from the attr name.

It seems _marginally_ better to my eyes though, so if status quo is the way to go, then `rust_` SGTM.

Reply:
IMHO, unless someone presents a slam dunk option, I’m inclined to leave this as-is and save the bikeshedding for a future iteration, perhaps when we expand the list of codegen options we support.

Otherwise, we’re iterating on names none of which seem significantly better than the others.